ngraph/frontend/paddlepaddle/src/op/pool2d.cpp

```diff
@@ -25,47 +15,232 @@ namespace ngraph
+                                    ngraph::Shape{0, 0}, // FIXME pads_begin
+                                    ngraph::Shape{0, 0}, // pads_end
+                                    ngraph::Shape{input_h, input_w})},
+                                {"Out"});
+                        }
+                        else
+                        {
+                            auto axes = ngraph::opset6::Constant::create(
+                                ngraph::element::i64, {2}, {input_rank - 2, input_rank - 1});
+                            return node.default_single_output_mapping(
+                                {std::make_shared<ngraph::opset6::ReduceMean>(data, axes, true)},
+                                {"Out"});
+                        }
+                    }
+                    else if (adaptive)
+                    {
+                        uint64_t pool_size_Height, pool_size_Width;
+
+                        if (kernel_shape.size() == 1)
+                        {
+                            // Not tested: implemented according to spec, but can't generate real
+                            // model to test
+                            pool_size_Height = pool_size_Width = kernel_shape[0];
+                        }
+                        else
+                        {
+                            pool_size_Height = kernel_shape[0];
+                            pool_size_Width = kernel_shape[1];
+                        }
+
+                        uint64_t stride_h = int64_t(input_h / pool_size_Height);
+                        uint64_t stride_w = int64_t(input_w / pool_size_Width);
+                        uint64_t kernel_h = input_h - (pool_size_Height - 1) * stride_h;
+                        uint64_t kernel_w = input_w - (pool_size_Width - 1) * stride_w;
+
+                        PDPD_ASSERT(stride_h >= 1 && stride_w >= 1,
+                                    "Pool2d stride must be greater than 1");
+
+                        if (pooling_type == "max")
+                        {
+                            return node.default_single_output_mapping(
+                                {std::make_shared<ngraph::opset6::MaxPool>(
+                                    data,
+                                    ngraph::Strides{stride_h, stride_w},
+                                    pad_begin,
+                                    pad_end,
+                                    ngraph::Shape{kernel_h, kernel_w},
+                                    rounding_type,
+                                    auto_pad)},
+                                {"Out"});
+                        }
+                        else
+                        {
+                            bool exclude_pad = node.get_attribute<bool>("exclusive") ? true : false;
+                            return node.default_single_output_mapping(
+                                {std::make_shared<ngraph::opset6::AvgPool>(
```

Comment:
Does it work correctly? Why we need AdaptiveAvgPool op?

Reply:
The `pdpd.fluid.layers.adaptive_pool2d` are needed by PDPD models. The adaptive means we the shape of filter for pool are computed by the input and output size of this pool2d convert. We have test this part using models generated by ngraph/test/files/paddlepaddle/gen_scripts/generate_pool2d.py.

Reply:
One more comment about adative pool2d. This kind of adaptive pool can only handle if the input size is multiple of output_size. e. g. input[8,8], out_put[4,4]. That's why a new Op adative_pool2d is planned.